app/helpers/utils.py

```diff
@@ -177,3 +178,30 @@ def get_registered_method(app, url_rule):
             ]
         )
     )
+
+
+def gzip_string(string):
+    try:
+        data = string.encode('utf-8')
+    except (UnicodeDecodeError, AttributeError) as error:
+        logger.error("Error when encoding string: %s", error)
+
+    gzipped_data = gzip.compress(data, compresslevel=5)
```

Comment:
`gzip.compress` requires a byte object otherwise it raise a `TypeError`.

Reply:
I would suggest to put this method directly in the `validate_kml()` method in such a way that his one returns always a gzip kml content. And if the kml cannot be gzip then return a 400 Bad Request.

Reply:
> `gzip.compress` requires a byte object otherwise it raise a `TypeError`.

Good point,  I'll put the `gzipped_data = gzip.compress(data, ...)` in a try-except and will except `TypeError`.

Reply:
Now this error is already intercepted in `validate_kml_file()`